### ISSUE TITLE
How to set optioal tlv while submit_sm

### ISSUE THREAD

**Issue by srimannarayanab** [python-smpplib/python-smpplib#204]
For sending messages in Submit_SM using SMPP API, there will be Two mandatory
parameters introduced in SMPP PDU.
These new tags are introduced in the optional parameter list. 
Optional Parameter Tag value in Hex Tag Value in Decimal
PE_ID 1400 5120
Template_ID 1401 5121

**Comment by gunjanbagde** [comment]
@eigenein @baverman @code-of-kpp @yusufk 
Anyone please answer this.

**Comment by code-of-kpp** [comment]
I'm not sure if I follow.
Have you tried specifying `allow_unknown_opt_params=True` when creating a client?

**Comment by gunjanbagde** [comment]
Yes I've  tried specifying allow_unknown_opt_params=True, and defining optional_params as dictionary and passing it in send.message method. as below

`optional_params = {
    5121: '110100001341',
    5120: '1107165409153095415',
}

pdu = smpp_client.send_message(
        source_addr_ton=consts.SMPP_TON_INTL,
        # source_addr_npi=smpplib.consts.SMPP_NPI_ISDN,
        # Make sure it is a byte string, not unicode:
        source_addr='XXXXX',

        dest_addr_ton=consts.SMPP_TON_INTL,
        # dest_addr_npi=smpplib.consts.SMPP_NPI_ISDN,
        # Make sure thease two params are byte strings, not unicode:
        destination_addr='no.',
        short_message=part,

        data_coding=encoding_flag,
        esm_class=msg_type_flag,
        registered_delivery=True,
        optional_params=optional_params
    )
`

**Comment by itsolution4india** [comment]
Have you Found any solution, i am also facing same issue

**Comment by yusufk** [comment]
Have you tried tracing on the wire using Wireshark to see what's happening?

**Comment by itsolution4india** [comment]
When I submit the send_message request, the SMPP provider receives all the required details except for entity_id and template_id. These parameters are included in the optional_parameters, but for messaging in India, both entity_id and template_id are mandatory. It seems they are not being transmitted correctly.

Additionally, I've tried sending messages to multiple recipients. For some numbers, I received the following error:
Received error code 901: <smpplib.command.SubmitSMResp object at 0x000001FFE4AFA210>
And for others, I got a delivery report (DLR) with the following status:
Received DLR: Status=UNDELIV, Full Response=id:940212558477101-68fc32dc sub:001 dlvrd:000 submit date:250405111722 done date:250405111722 stat:UNDELIV err:442 text:
Please help troubleshoot why the optional parameters are not being delivered correctly, and what these errors might indicate.
